mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresMailboxDAO.java

```diff
@@ -140,4 +143,38 @@ private Mailbox asMailbox(Record record) {
         return new Mailbox(new MailboxPath(record.get(MAILBOX_NAMESPACE), Username.of(record.get(USER_NAME)), record.get(MAILBOX_NAME)),
             UidValidity.of(record.get(MAILBOX_UID_VALIDITY)), PostgresMailboxId.of(record.get(MAILBOX_ID)));
     }
+
+    public Mono<ModSeq> findHighestModSeqByMailboxId(MailboxId mailboxId) {
```

Comment:
Just to be very clear:

We will get a refactoring quite soon to reduce DB hits as part of IMAP STATUS command.

So the way I see it:

Introduce a `PostgresMailbox` POJO that extends `SimpleMailbox` POJO. This POJO adds `lastUid` and `nextModSeq` metadata as part of the mailbox. We could then override the `PostresMessageManager::getMetadata` method to leverage this and cut on DB requests.

Thoughts?

(Maybe follow up tickets but this is the kind of crap I like doing ;-) )

So that part of the SELECT / STATUS command is just about "casting the PostgresMailbox"`

Reply:
https://github.com/linagora/james-project/issues/4945